mongo/integration/retryable_writes_prose_test.go

```diff
@@ -137,4 +140,77 @@ func TestRetryableWritesProse(t *testing.T) {
 			assert.NotNil(mt, err, "expected no txnNumber, got %v", txnNumber)
 		})
 	})
+
+	tpm := newTestPoolMonitor()
+	// Client options with MaxPoolSize of 1 and RetryWrites used per the test description.
+	// Lower HeartbeatInterval used to speed the test up for any server that uses streaming
+	// heartbeats. Only connect to first host in list for sharded clusters.
+	hosts := mtest.ClusterConnString().Hosts
+	pceOpts := options.Client().SetMaxPoolSize(1).SetRetryWrites(true).
+		SetPoolMonitor(tpm.PoolMonitor).SetHeartbeatInterval(500 * time.Millisecond).
+		SetHosts(hosts[:1])
+
+	mtPceOpts := mtest.NewOptions().ClientOptions(pceOpts).MinServerVersion("4.3").
+		Topologies(mtest.ReplicaSet, mtest.Sharded)
+	mt.RunOpts("PoolClearedError retryability", mtPceOpts, func(mt *mtest.T) {
```

Comment:
Optional: Consider adding a comment that references `GODRIVER-1910` and/or the prose test that this implements.

Reply:
Will leave this to GODRIVER-2274.